### PR TITLE
Change PyTorch noisy demo to use Amazon Braket

### PR DESCRIPTION
**Pull request by antalszava**
[PennyLane-Forest has discontinued support for the Rigetti QPU](https://github.com/PennyLaneAI/pennylane-forest/pull/68), this PR removes the reference to that device.

### CODE REVIEW

File: demonstrations/pytorch_noise.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
    tutorial_noisy_circuit_optimization Optimizing noisy circuits with Cirq
 
-*Author: PennyLane dev team. Last updated: 1 Mar 2020.*
+*Author: PennyLane dev team. Last updated: 16 Jun 2020.*
```
Comment (co9olguy) [inline demonstrations/pytorch_noise.py:16]:
```suggestion
*Author: PennyLane dev team. Last updated: 16 Jun 2021.*
```

**Comment by co9olguy** [comment]
We might want to consider more than a single-line change to this demo, since it seems specifically written for QPU examples, e.g., "Extend PyTorch with real quantum computing power, by using it to optimize a noisy quantum hardware device."

File: demonstrations/pytorch_noise.py [hunk 2]
```diff
@@ -173,16 +173,32 @@ def cost(phi, theta, step):
 #
 # As PyTorch natively supports GPU-accelerated classical processing, and Forest provides
 # quantum hardware access in the form of QPUs, with very little modification, we can run
-# the above code as a hybrid GPU-QPU optimization (note that to run the following
-# script, you will need to be using Rigetti's QCS service):
+# the above code as a hybrid GPU-QPU optimization.
+#
+# Note that to run the following script, you will need access to Rigetti's QPU.
```
Comment (co9olguy) [inline demonstrations/pytorch_noise.py:178]: maybe also update the reference to Forest here

File: demonstrations/pytorch_noise.py [hunk 3]
```diff
@@ -173,16 +173,32 @@ def cost(phi, theta, step):
 #
 # As PyTorch natively supports GPU-accelerated classical processing, and Forest provides
```
Comment (co9olguy) [inline demonstrations/pytorch_noise.py:174]: whuggins@google.com

File: demonstrations/pytorch_noise.py [hunk 4]
```diff
@@ -13,7 +13,7 @@
 
    tutorial_noisy_circuit_optimization Optimizing noisy circuits with Cirq
```
Comment (co9olguy) [inline demonstrations/pytorch_noise.py:14]: nathan+test@xanadu.ai

File: demonstrations/pytorch_noise.py [hunk 5]
```diff
@@ -171,18 +171,34 @@ def cost(phi, theta, step):
 # Hybrid GPU-QPU optimization
 # ---------------------------
 #
-# As PyTorch natively supports GPU-accelerated classical processing, and Forest provides
-# quantum hardware access in the form of QPUs, with very little modification, we can run
-# the above code as a hybrid GPU-QPU optimization (note that to run the following
-# script, you will need to be using Rigetti's QCS service):
+# As PyTorch natively supports GPU-accelerated classical processing, and Amazon
+# Braket provides quantum hardware access in the form of QPUs, with very little
+# modification, we can run the above code as a hybrid GPU-QPU optimization.
```
Comment (albi3ro) [inline demonstrations/pytorch_noise.py:176]:
```suggestion
# Braket provides quantum hardware access in the form of QPUs, we can run the above code 
# as a hybrid GPU-QPU optimization with very little modifications.
```

**Review by albi3ro — COMMENTED** [review]
First I was confused and thought everything forest related was deprecated, then I realized it was just QPU access.

I think out of the gate we should start saying we are using the Rigetti's ``forest.qvm`` simulator and comparing it to Rigetti's AWS Braket QPU.

Also, is `forext.qvm` a GPU simulator? The first time you mention GPU is during the hybrid optimization paragraph.  

A big bottleneck with GPU calculations is getting data to and from the GPU. If the data has to go to and from a remote QPU each time, what's the benefit of running it on a GPU? I'm not convinced.

**Comment by antalszava** [comment]
Thanks for the comments!

> I think out of the gate we should start saying we are using the Rigetti's forest.qvm simulator and comparing it to Rigetti's AWS Braket QPU.

Added a mention at the beginning.

> Also, is forext.qvm a GPU simulator? The first time you mention GPU is during the hybrid optimization paragraph.

There is a mention of the QVM at the beginning of the tutorial. Torch provides the GPU support.

> A big bottleneck with GPU calculations is getting data to and from the GPU. If the data has to go to and from a remote QPU each time, what's the benefit of running it on a GPU? I'm not convinced.

Sounds reasonable, perhaps this could be a reason to reconsider the content of the demo.

File: demonstrations/pytorch_noise.py [hunk 6]
```diff
@@ -13,12 +13,14 @@
...
+*Author: PennyLane dev team. Last updated: 16 Jun 2021.*
 
 Let's revisit the original :ref:`qubit rotation <qubit_rotation>` tutorial, but instead of
 using the default NumPy/autograd QNode interface, we'll use the :doc:`introduction/interfaces/torch`.
-We'll also replace the ``default.qubit`` device with a noisy ``forest.qvm`` device, to
-see how the optimization responds to noisy qubits.
+We'll also replace the ``default.qubit`` device with a noisy ``forest.qvm``
+device, to see how the optimization responds to noisy qubits. At the end of the
+demonstration, we will also show a way of how Rigetti's QPU can be used via
+Amazon Braket.
```
Comment (albi3ro) [inline demonstrations/pytorch_noise.py:23]: 👍

File: demonstrations/pytorch_noise.py [hunk 7]
```diff
@@ -171,18 +181,34 @@ def cost(phi, theta, step):
...
-# script, you will need to be using Rigetti's QCS service):
+# As PyTorch natively supports GPU-accelerated classical processing, and Amazon
+# Braket provides quantum hardware access in the form of QPUs, we can run the above code 
+# as a hybrid GPU-QPU optimization with very little modification.
+#
+# Note that to run the following script, you will need access to Rigetti's QPU.
+# To connect to a QPU, we can use Amazon Braket. For a dedicated demonstration
+# on using Amazon Braket, see our tutorial on [Computing gradients in parallel
+# with Amazon
+# Braket](https://pennylane.ai/qml/demos/braket-parallel-gradients.html).
```
Comment (albi3ro) [inline demonstrations/pytorch_noise.py:192]:
```suggestion
# on using Amazon Braket, see our tutorial on
# `Computing gradients in parallel with Amazon Braket <https://pennylane.ai/qml/demos/braket-parallel-gradients.html>`_
```
This link needs to be in rst form, not markdown form.

**Review by albi3ro — CHANGES_REQUESTED** [review]
Just need to fix the link that's not rendering properly, and then I'll approve :)

**Comment by antalszava** [comment]
@albi3ro good catch, thanks! Updated

**Review by albi3ro — APPROVED** [review]
Good improvement :)

**Pull request merged**